### PR TITLE
feat: log option parameter

### PR DESCRIPTION
**Pull request by romanbalayan**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -37,7 +37,18 @@ export const createAppAuth: StrategyInterface = function createAppAuth(
...
       ? { installationId: Number(options.installationId) }
-      : {}
+      : {},
+    {
+      log: Object.assign(
+        {
+          debug: () => {},
+          info: () => {},
+          warn: console.warn.bind(console),
+          error: () => {},
```
Comment (gr2m) [inline src/index.ts:47]: as we only use `log.warn`, I'd simply remove the other methods 
```suggestion
          warn: console.warn.bind(console),
```

File: test/index.test.ts [hunk 2]
```diff
@@ -1831,3 +1836,66 @@ test("id and installationId can be passed as options", async () => {
...
+    });
+
+  const auth = createAppAuth({
+    id: APP_ID,
+    privateKey: PRIVATE_KEY,
+    log: {
+      debug: () => calls.push("debug"),
+      info: () => calls.push("info"),
+      warn: () => calls.push("warn"),
+      error: () => calls.push("error"),
```
Comment (gr2m) [inline test/index.test.ts:1878]:
```suggestion
      warn: () => calls.push("warn"),
```

**Review by gr2m — APPROVED** [review]
This looks great, it only requires a small change to get to the 100% code coverage. Thank you 💐 

Because I show this live on raise.dev/hacktoberfest, I'll go ahead and do the changes myself so I can show some other things :)

File: src/types.ts [hunk 3]
```diff
@@ -82,6 +82,13 @@ export type StrategyOptions = {
   request?: OctokitTypes.RequestInterface;
   cache?: Cache;
   timeDifference?: number;
+  log?: {
+    debug: (message: string, additionalInfo?: object) => any;
+    info: (message: string, additionalInfo?: object) => any;
+    warn: (message: string, additionalInfo?: object) => any;
+    error: (message: string, additionalInfo?: object) => any;
```
Comment (gr2m) [inline src/types.ts:89]:
```suggestion
    warn: (message: string, additionalInfo?: object) => any;
```

File: src/types.ts [hunk 4]
```diff
@@ -121,4 +125,11 @@ export type State = StrategyOptions & {
   installationId?: number;
   request: OctokitTypes.RequestInterface;
   cache: Cache;
+  log: {
+    debug: (message: string, additionalInfo?: object) => any;
+    info: (message: string, additionalInfo?: object) => any;
+    warn: (message: string, additionalInfo?: object) => any;
+    error: (message: string, additionalInfo?: object) => any;
```
Comment (gr2m) [inline src/types.ts:132]:
```suggestion
    warn: (message: string, additionalInfo?: object) => any;
```

**Comment by romanbalayan** [comment]
Hey @gr2m!

Watched the whole twitch video, and thank you for the shout out! Super interesting workflow you have on how you are maintaining Octokit. I'm a big fan of automation myself, and the process you have is really insightful. Hopefully, I get to contribute on other Octokit projects as well! Thanks a lot!

**Pull request merged**